client/errors.go

```diff
@@ -16,37 +16,76 @@ import (
 func ErrorClassifier(meta schema.ClientMeta, resourceName string, err error) diag.Diagnostics {
 	client := meta.(*Client)
 
+	resIdOverride := func(_ *diag.BaseError) {} // no-op option
+
+	if d, ok := err.(diag.Diagnostic); ok && len(d.Description().ResourceID) > 0 {
+		resIdList := []string{
+			client.AccountID,
+			client.Region,
+		}
+
+		// remove accountID and region from PK list as we always prepend them
+		for _, val := range d.Description().ResourceID {
+			if val != client.AccountID && val != client.Region {
+				resIdList = append(resIdList, val)
+			}
+		}
+
+		resIdOverride = diag.WithResourceId(resIdList)
+	}
```

Comment:
nit: maybe make this a function? and pass a WithAWSResoruceId(d) (do the logic you did here) and then you can call this function in each return making this function lighter

Reply:
fixed